### DISCUSSION TITLE
Sofar2mqtt 2.0b1

### DISCUSSION THREAD

**Discussion by double1968** [cmcgerty/Sofar2mqtt#22] [Q&A]
I have upgraded to version 2.0
everything works and the system seems more stable.
I don't understand why I send the commands to my HYD4000 sometimes I receive in the topic "Sofar2mqtt/response/<function>" the value 512 as an answer while the notes report this:

The result code will be 0 for success, 1 means "Invalid Work Mode" (which possibly means
the inverter isn't in passive mode,) and 3 means "Inverter busy." 2 and 4 are data errors
which shouldn't happen unless there's a cable issue or some such.

Many Thanks

**Comment by Ernest-burgess** [comment]
In the notes it also states you need to. AND. That response with oxff to get the 0 response 
In my calculater 512 And 255 = 0.  Hope this helps

**Comment by chaosnature** [comment]
![image](https://github.com/cmcgerty/Sofar2mqtt/assets/52825000/c605cdbb-8acc-4c30-8979-2934a17fa354)
